src/Tmds.Systemd.Logging/JournalLogger.cs

```diff
@@ -100,15 +102,22 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     }
                     if (exception != null)
                     {
-                        logMessage.Append(Exception, exception.Message);
-                        logMessage.Append(ExceptionType, exception.GetType().FullName);
-                        logMessage.Append(ExceptionStackTrace, exception.StackTrace);
-                        Exception innerException = exception.InnerException;
-                        if (innerException != null)
+                        if (_exceptionFormatter != null)
                         {
-                            logMessage.Append(InnerException, innerException.Message);
-                            logMessage.Append(InnerExceptionType, innerException.GetType().FullName);
-                            logMessage.Append(InnerExceptionStackTrace, innerException.StackTrace);
+                            _exceptionFormatter(exception, logMessage);
+                        }
+                        else
+                        {
```

Comment:
Expose the `else` branch as `internal static readonly Action<Exception, JournalMessage> DefaultExceptionFormatter`.